.github/workflows/build_deploy.yml

```diff
@@ -143,7 +144,7 @@ jobs:
   upload_pypi:
     needs: [build_wheels_x86_64, build_wheels_i686, build_wheels_aarch64, build_sdist]
     runs-on: ubuntu-latest
-    if: github.event_name == 'release' && github.event.action == 'published'
+    if: (github.event_name == 'release' && github.event.action == 'published') || (github.event.action == 'workflow_dispatch')
```

Comment:
```suggestion
    if: (github.event_name == 'release' && github.event.action == 'published') && (github.event.action == 'workflow_dispatch')
```

wonder if we should make this an && in case we accidentally publish a release with the wrong tags. Thoughts @DataDog/apm-python?

Reply:
@Kyle-Verhoog so releasing to pypi would require the manual dispatch to be triggered as well as the release to be published?

Reply:
I don't think this can be an `&&` since we could never trigger a job manually and from publishing a release at the same time.


Depending on how this works, we could go the other way and remove the trigger on release publish and instead make it manual only, but manually it'll create a GH release for us ?

Reply:
Ah true. 

Yeah the way we trigger it now is way too error prone at the moment 
![image](https://user-images.githubusercontent.com/6321485/115071952-ebb1aa80-9ec4-11eb-9a28-a7451e03d5fa.png)

Reply:
This is tangential for this PR however, so :+1: from me